tests/projects/test_project.py

```diff
@@ -1464,3 +1463,189 @@ def test_load_project_from_yaml_with_function(context):
+                "authenticationMode": "none",
+                "upstreams": [
+                    {"kind": "nucliofunction", "nucliofunction": {"name": "fff"}}
+                ],
+            },
+            "status": {"name": "gateway-f1-f2", "state": "ready"},
+        }
+    ]
+    project_name = "project-name"
+    project = mlrun.new_project(project_name, context=str(context), save=False)
+    f1 = mlrun.code_to_function(
+        name="my-func1",
+        image="my-image",
+        kind=kind_1,
+        filename=str(assets_path() / "handler.py"),
+    )
+    f1.save()
+    functions = f1
+    project.set_function(f1)
+    if kind_2:
+        f2 = mlrun.code_to_function(
+            name="my-func2",
+            image="my-image",
+            kind=kind_2,
+            filename=str(assets_path() / "handler.py"),
+        )
+        f2.save()
+        project.set_function(f2)
+        functions = [f1, f2]
+
+    gateway = project.create_api_gateway(
+        name="gateway-f1-f2", functions=functions, canary=canary
+    )
+
+    assert (
+        gateway._invoke_url
+        == "gateway-f1-f2-project-name.default-tenant.app.dev.lab.iguazeng.com/"
+    )
+
+    assert gateway.authentication_mode == "none"
+
+
+@pytest.mark.parametrize(
+    "kind_1 ,kind_2, canary",
+    [
+        ("nuclio", "nuclio", [20]),
+        ("nuclio", "nuclio", [20, 10]),
+        ("nuclio", "job", [20, 80]),
+        ("job", None, None),
+    ],
+)
+@unittest.mock.patch.object(mlrun.db.nopdb.NopDB, "create_api_gateway")
+def test_create_api_gateway_invalid(
+    patched_create_api_gateway, context, kind_1, kind_2, canary
+):
+    patched_create_api_gateway.return_value = True
+    project_name = "project-name"
+    project = mlrun.new_project(project_name, context=str(context), save=False)
+    f1 = mlrun.code_to_function(
+        name="my-func1",
+        image="my-image",
+        kind=kind_1,
+        filename=str(assets_path() / "handler.py"),
+    )
+    f1.save()
+    functions = f1
+    project.set_function(f1)
+    if kind_2:
+        f2 = mlrun.code_to_function(
+            name="my-func2",
+            image="my-image",
+            kind=kind_2,
+            filename=str(assets_path() / "handler.py"),
+        )
+        f2.save()
+        project.set_function(f2)
+        functions = [f1, f2]
+
+    with pytest.raises(mlrun.errors.MLRunInvalidArgumentError):
+        project.create_api_gateway(
+            name="gateway-f1-f2", functions=functions, canary=canary
+        )
+
+
+@unittest.mock.patch.object(mlrun.db.nopdb.NopDB, "list_api_gateways")
+def test_list_api_gateways(patched_list_api_gateways, context):
+    patched_list_api_gateways.return_value = [
+        {
+            "metadata": {
+                "name": "test",
+                "namespace": "default-tenant",
+                "labels": {
+                    "iguazio.com/username": "admin",
+                    "nuclio.io/project-name": "default",
+                },
```

Comment:
keep the mocks simple, non iguazio specific.
i.e. remove username label and `default-tenant`